src/pages/transactions/index.tsx

```diff
@@ -1,3 +1,25 @@
+import { useEffect } from 'react'
+import type { NextPage } from 'next'
+import { useRouter } from 'next/router'
+import { AppRoutes } from '@/config/routes'
+import useTxQueue from '@/hooks/useTxQueue'
 import HistoryPage from './history'
 
-export default HistoryPage
+const TransactionsPage: NextPage = () => {
+  const router = useRouter()
+  const { page } = useTxQueue()
+  const hasQueuedTxs = !!page?.results.length
+
+  useEffect(() => {
+    const pathname = hasQueuedTxs ? AppRoutes.transactions.queue : AppRoutes.transactions.history
+
+    router.push({
+      pathname,
+      query: { safe: router.query.safe },
+    })
+  }, [router, hasQueuedTxs])
```

Comment:
Can this be achieved in the `return`? Currently, if the user is on the history page and a transaction is added to the queue, they will be redirected without user input.

Reply:
They will be redirected only once to queue or history, and this effect won't run on those pages.